ports/servoshell/webview.rs

```diff
@@ -250,6 +271,65 @@ where
         }
     }
 
+    fn play_haptic_effect(&mut self, index: usize, params: DualRumbleEffectParams) {
+        if let Some(ref mut gilrs) = self.gamepad {
+            if let Some(connected_gamepad) = gilrs
+                .gamepads()
+                .find(|gamepad| usize::from(gamepad.0) == index)
+            {
+                let start_delay = Ticks::from_ms(params.start_delay as u32);
+                let duration = Ticks::from_ms(params.duration as u32);
+                let strong_magnitude = (params.strong_magnitude * u16::MAX as f64).round() as u16;
+                let weak_magnitude = (params.weak_magnitude * u16::MAX as f64).round() as u16;
+
+                let scheduling = Replay {
+                    after: start_delay,
+                    play_for: duration,
+                    with_delay: Ticks::from_ms(0),
+                };
+                let effect = EffectBuilder::new()
+                    .add_effect(BaseEffect {
+                        kind: BaseEffectType::Strong { magnitude: strong_magnitude },
+                        scheduling,
+                        envelope: Default::default(),
+                    })
+                    .add_effect(BaseEffect {
+                        kind: BaseEffectType::Weak { magnitude: weak_magnitude },
+                        scheduling,
+                        envelope: Default::default(),
+                    })
+                    .repeat(Repeat::For(start_delay + duration))
+                    .add_gamepad(&connected_gamepad.1)
+                    .finish(gilrs)
+                    .expect("Failed to create haptic effect, ensure connected gamepad supports force feedback.");
+                self.haptic_effects.as_mut_slice()[index] = Some(effect);
+                self.haptic_effects[index]
+                    .as_ref()
+                    .expect("No haptic effect found")
+                    .play()
+                    .expect("Failed to play haptic effect.");
+            } else {
+                println!("Couldn't find connected gamepad to play haptic effect on");
+            }
+        }
+    }
+
+    fn stop_haptic_effect(&mut self, index: usize) {
+        let Some(ref effect) = self.haptic_effects[index] else {
+            return;
+        };
+
+        effect.stop().expect("Failed to stop haptic effect.");
```

Comment:
can we handle this in some way?

Reply:
Maybe for the "was succesffuly stopped" thing the in the reply(see above comment).